modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -1174,10 +1176,11 @@ private void handleSortByClause(final Query query) {
                 }
 
                 // assign to query if sorting is supported!
-                final SortBy[] sb = clauses.toArray(new SortBy[] {});
+
+                this.sortBy = clauses.toArray(new SortBy[] {});
                 if (rasterManager.getGranuleCatalog()
-                        .getQueryCapabilities(rasterManager.getTypeName()).supportsSorting(sb)) {
-                    query.setSortBy(sb);
+                        .getQueryCapabilities(rasterManager.getTypeName()).supportsSorting(sortBy)) {
+                    query.setSortBy(sortBy);
```

Comment:
In case of heterogeneous CRS I'd add the crs property as a last sort-by property, so that we don't end up creating sub-mosaics for nothing (mosaicking is an expensive operation, we should try to do as few as possible within the respect of the user request)